homeassistant/components/hydrawise/coordinator.py

```diff
@@ -55,11 +57,29 @@ async def _async_update_data(self) -> HydrawiseData:
                 zones[zone.id] = zone
             for sensor in controller.sensors:
                 sensors[sensor.id] = sensor
-            daily_water_summary[controller.id] = await self.api.get_water_use_summary(
-                controller,
-                now().replace(hour=0, minute=0, second=0, microsecond=0),
-                now(),
-            )
+
+            usage_summary: ControllerWaterUseSummary | None = None
+            if (
+                self._last_usage_refresh is None
+                or _utcnow - self._last_usage_refresh > WATER_USAGE_SCAN_INTERVAL
```

Comment:
This will actually break the usage of the homeassistant.update_entity service. Honestly, I think the best solution would be the one that is suggested by Joost above; by putting this in a different coordinator.

Reply:
It's less broken than it is now, and for a very specific edge-case (how often do you think people are actually calling the update_entity service? I would suspect not very often...). 

Again, I'm happy to follow up with a better solution later, but IMO fixing the existing brokenness seems like a higher priority than getting this perfect.

Reply:
Found more time today than I thought I had and finished the refactor.